cou/commands.py

```diff
@@ -153,6 +190,25 @@ def get_subcommand_common_opts_parser() -> argparse.ArgumentParser:
         type=batch_size_arg,
         default=1000,
     )
+    subcommand_common_opts_parser.add_argument(
+        "--purge",
+        help="Delete data from shadow tables. before cloud upgrade.\n" "Default to disable purge.",
+        action="store_true",
+        default=False,
+    )
+    subcommand_common_opts_parser.add_argument(
+        "--purge-before-date",
+        dest="purge_before",
+        action=PurgeBeforeArgumentAction,
+        help=(
+            "Providing this argument will delete data from all shadow tables"
+            "\nthat is older than the date provided."
+            "\nDate string format should be YYYY-MM-DD[HH:mm][:ss]."
+            "\nWithout --purge-before the purge step will delete all the data."
+        ),
+        type=purge_before_arg,
```

Comment:
Yes, you are right, it's because I created `purge_before_arg` first then add `PurgeBeforeArgumentAction` later but I don't merge them together.

I can provide another refactor PR later.